### DISCUSSION TITLE
[Beginner] Authorized route with folder bugs out?

### DISCUSSION THREAD

**Discussion by Ookamini95** [TanStack/router#1587] [Q&A] [answered]
https://tanstack.com/router/latest/docs/framework/react/guide/authenticated-routes#non-redirected-authentication

I'm using the above snippet to make auth, my folder structure is 

```
index.tsx
_private
   >_auth.tsx
   >_auth.profile.tsx
```
    
this results in a weird `/_private/_auth` situation. This is due me trying (and failing) to make it work with
```
index.tsx
_auth
    >index.tsx
    profile
        >index.tsx
```
and not working since this makes another '/' that conflicts with base index

**Comment by SeanCassiere** [comment, marked as answer]
Would probably need an understanding of the URLs you are looking build, but in essence this is what should be getting created.

```
src/routes

index.tsx - resolves to localhost:3000/
_private.tsx - layout route configuration
_private/
    _auth.tsx - layout route configuration
    _auth.profile.tsx - resolves to localhost:3000/profile
```

The `_private.tsx` and `_auth.tsx` should have whatever logic and components configured in them, with an `<Outlet />` rendered where the child route should be shown.

All in all, only the following URLs would be created: `localhost:3000/` and `localhost:3000/profile`.

**Reply by Ookamini95** [reply]
I'm sorry in the end our team is in a bit of a rush so we decided to go for another solution.
Atm I'm crunching and have no free time (and soul) but will try tanstack router in a later project.
Thanks for answering, will mark your answer as the correct answer assuming you were correct.
